locked-asset/proxy_dex/src/lib.rs

```diff
@@ -36,6 +36,8 @@ pub trait ProxyDexImpl:
     + events::EventsModule
     + elrond_wasm_modules::default_issue_callbacks::DefaultIssueCallbacksModule
     + utils::UtilsModule
+    + locking_module::locking_module::LockingModule
```

Comment:
You should use `LockWithEnergyModule`, not `locking_module`, that's for simple-lock. `LockWithEnergyModule` is for energy-factory.

Reply:
Add method for locking, similar to the one from `LockingModule`, currently, the `LockWithEnergyModule` is only used for virtual locking.